mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -69,70 +152,216 @@ def subset_variables(ds, vlist):  # {{{
     # drop spurious coordinates
     ds = ds.drop(dropcoords)
 
-    assert len(ds.variables.keys()) > 0, \
-        'MPAS_XARRAY ERROR: Empty dataset is returned.\n' \
-        'Variables {}\nare not found within the dataset ' \
-        'variables: {}.'.format(vlist, allvars)
+    if len(ds.data_vars.keys()) == 0:
+        raise ValueError(
+                'Empty dataset is returned.\n'
+                'Variables {}\n'
+                'are not found within the dataset '
+                'variables: {}.'.format(variable_list, allvars))
+
+    return ds  # }}}
+
+
+def preprocess(ds, time_variable_name, variable_list, sel_values,
+               isel_values, year_offset):  # {{{
+    """
+    Builds correct time specification for MPAS, allowing a date offset
+    because the time must be between 1678 and 2262 based on the xarray
+    library.  Also, if slicing information (`sel_values` and/or
+    `isel_values`) was provided in `openMultifileDataSet`, this
+    function performs the appropriate slicing on the data set.
+
+    Parameters
```

Comment:
Do we know that these current doc strings will be parse-able via sphinx?  I don't think we need to verify this now as long as we are willing to make tweaks to the docstrings later when we build out a automatic doc generation capability.